feat: Add support for the `jest.config.ts` configuration file (#10564)
facebook/jest: docs/Configuration.md

};
```

Or through TypeScript:

```ts
// jest.config.ts
import type { Config } from '@jest/types';

//Sync object

> export = {
>   verbose: true,
> } as Config.InitialOptions;

Reviewer: ```suggestion
const config: Config.InitialOptions = {
  verbose: true
};

export = config;
```

Casting is subtly different from annotating: it'll have the compiler warn for excess properties but not for missing required properties, including with nested objects.

While I think most of our options (if not all) are optional, I think it's better to encourage using an annotation (see #9570 & co for examples)- keep in mind that you can use module augmentation to include properties from other packages like `ts-jest` which might have required properties.

Author: @G-Rath Make sense, thank you for input. I have some tests to fix and I will push a commit with this changes.

};
```

Or through TypeScript (if `ts-node` is installed):

```ts
// jest.config.ts
import type {Config} from '@jest/types';

> //Sync object

Reviewer 2: ```suggestion
// Sync object
```

const config: Config.InitialOptions = {
  verbose: true,
};
export default config;

> //Or async function

Reviewer 2: ```suggestion
// Or async function
```

};
```

> Or through TypeScript:
> 
> You can enable support for the TypeScript configuration file by installing the `ts-node` plugin in your project.

Reviewer: ```suggestion
Or through TypeScript (if `ts-node` is installed):
```

nit: I think this flows a bit better.
